searchcore/src/vespa/searchcore/proton/server/documentdb.cpp

```diff
@@ -517,15 +508,25 @@ DocumentDB::performDropFeedView(IFeedView::SP feedView)
 
 
 void
-DocumentDB::performDropFeedView2(IFeedView::SP feedView)
-{
+DocumentDB::performDropFeedView2(IFeedView::SP feedView) {
     // Called by executor task, delays when feed view is dropped.
     // Also called by DocumentDB::receive() method to keep feed view alive
     _writeService.indexFieldInverter().sync();
     _writeService.indexFieldWriter().sync();
+    masterExecute([this, feedView]() { performDropFeedView3(feedView, 10); });
+}
 
-    // Feed view is kept alive in the closure's shared ptr.
-    masterExecute([feedView] () { doNothing(feedView); });
+void
+DocumentDB::performDropFeedView3(IFeedView::SP feedView, uint32_t numRetries) {
+    // We must keep the feedView allive until all operations are drained.
+    // TODO: This is a very brittle appraoch that we should reconsider.
+    if (feedView && ! feedView->isDrained()) {
+        LOG(warning, "FeedView for document type '%s' has not been drained. Reposting to check again. %d retries left",
+            getName().c_str(), numRetries);
+        if (numRetries > 0) {
+            masterExecute([this, feedView, numRetries]() { performDropFeedView3(feedView, numRetries - 1); });
```

Comment:
Consider adding `_writeService.sync();` before calling masterExecute(), to ensure each retry does a new sync of 
_writeService.

Reply:
Fixed